### ISSUE TITLE
NUM_PROC_THREADS not being honored when processing input.

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9234]
**Describe the bug**
In an 8-thread computer, I enable NUM_PROC_THREADS to 4 when building my documentation. However, in some steps, doxygen uses 8 even if I told him to only use 4. In other steps, it correctly respects the maximum.

**Expected behavior**
It always respects the maximum.

**Screenshots**
doxygen's output is very clear about this:
```
...
Processing input using 8 threads.
...
Generating code files using 4 threads.
...
```

**To Reproduce**
Just turn in any project NUM_PROC_THREADS to half of your max number of threads, and check the output.

**Version**
`1.9.4 (91095fb62f0bc2c624b0e8e49d9316f51be76559*)`

**Comment by ferdymercury** [comment]
I proposed a patch: https://github.com/doxygen/doxygen/pull/9235

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9237, which in my opinion supersedes the proposed pull request #9235

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
